### ISSUE TITLE
0 labels saved

### ISSUE THREAD

**Issue by barroca8** [ultralytics/ultralytics#1053]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I created a model using RoboFlow polygon labeled data. (YOLO, not pascalVOC)

I trained it:

```
model = YOLO('yolov8n-seg.yaml') #empty model

results = model.train(data='data.yaml', epochs=50, workers=1, batch=8, imgsz=640)
```

And I changed the model to be `best.pt` :

```
model = YOLO("/path_to_best/best.pt")
```
I tried to run the model on a folder of 61 images I have, but it's always saving the data unlabeled:

```
!yolo task=segment mode=predict model = /path_to_best/best.pt source = /path_to_test_set/test_set save=True save_txt=True
```
And this returns:
```
Ultralytics YOLOv8.0.40 🚀 Python-3.8.10 torch-1.13.1+cu116 CUDA:0 (Tesla T4, 15110MiB)
YOLOv8n-seg summary (fused): 195 layers, 3259429 parameters, 0 gradients, 12.0 GFLOPs

image 1/61 /path_to_test_set/test_set/1.jpg: 448x640 31.4ms
image 2/61 /path_to_test_set/test_set/10.jpeg: 448x640 24.6ms
```
(...)
```
image 61/61 /path_to_test_set/test_set/9.jpeg: 448x640 12.9ms
Speed: 0.6ms pre-process, 18.3ms inference, 0.5ms postprocess per image at shape (1, 3, 640, 640)
Results saved to runs/segment/predict4
0 label saved to runs/segment/predict4/labels
```
And when I open the predict folder (in this case predict4), all images have no labels.

What could possibly be causing this?

(Could it be because the images aren't 640x640?? But that wouldn't make sense because that would assume that the model can only process 640x640 images)

Note: This is the val_batch0_labels.jpg in the results -> train folder. The model was trained well, there shouldn't be a reason for this to happen

![image](https://user-images.githubusercontent.com/89426383/219974488-dca9caf8-caf7-4f23-b394-5794f0416afd.png)

**Comment by q465414859** [comment]
练的少 素材少

**Comment by Laughing-q** [comment]
@barroca8 
> Note: This is the val_batch0_labels.jpg in the results -> train folder. The model was trained well, there shouldn't be a reason for this to happen

well actually `val_batch0_labels.jpg` is showing labels and you should check `val_batch0_preds.jpg` which shows predictions.
From your description,  it seems the training is not going well so adding more training images could help you.

**Comment by glenn-jocher** [comment]
@barroca8 there are no detections on your images. This is probably just because your model has not trained sufficiently. You should train it much longer, i.e. for 300 or 1000 epochs if your dataset is small.

@Laughing-q I think we should add a "no detections" notice to the predict output so users are not confused

**Comment by glenn-jocher** [comment]
@Laughing-q @AyushExel fixed in https://github.com/ultralytics/ultralytics/pull/1069/commits/7707829f27f49a7992900052a74f626f79fcf509

Now `(no detections)` will be actively shown to actively inform the user that inference ran correctly and no objects were detected:
<img width="872" alt="Screenshot 2023-02-20 at 21 55 08" src="https://user-images.githubusercontent.com/26833433/220197188-d74fa459-0ad1-48a3-8e6f-b53a4f8f19c1.png">

**Comment by barroca8** [comment]
> @barroca8 there are no detections on your images. This is probably just because your model has not trained sufficiently. You should train it much longer, i.e. for 300 or 1000 epochs if your dataset is small.
> 
> @Laughing-q I think we should add a "no detections" notice to the predict output so users are not confused

Thank you so much! I can't find a way to run more than 100 epochs for free. Google Colab (free), with or without GPU, colapses after a while and has some general warning about "usage limits". Jupyter Notebook crashes as soon I start to train the model. What other free alternatives could allow me to train my model through 1000 epochs?
